artemis-server/src/main/java/org/apache/activemq/artemis/core/paging/cursor/impl/PageSubscriptionImpl.java

```diff
@@ -1350,7 +1350,7 @@ public synchronized boolean hasNext() {
             return true;
          }
 
-         if (!pageStore.isPaging()) {
+         if (!pageStore.isPagingDirtyRead()) {
```

Comment:
Concern here is this ins't an async case.

Btw i cannot see the change in QueueImpl to use the new paging dirty read.